<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>libsequence: Sequence::VariantMatrix Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">libsequence
   &#160;<span id="projectnumber">1.9.5</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',false,false,'search.php','Search');
});
</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('classSequence_1_1VariantMatrix.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="summary">
<a href="#pub-types">Public Types</a> &#124;
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="#pub-static-attribs">Static Public Attributes</a> &#124;
<a href="classSequence_1_1VariantMatrix-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">Sequence::VariantMatrix Class Reference<div class="ingroups"><a class="el" href="group__popgen.html">Molecular Population Genetics</a> &raquo; <a class="el" href="group__variantmatrix.html">Variant Matrix</a></div></div>  </div>
</div><!--header-->
<div class="contents">

<p>Matrix representation of variation data.  
 <a href="classSequence_1_1VariantMatrix.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="VariantMatrix_8hpp_source.html">VariantMatrix.hpp</a>&gt;</code></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-types"></a>
Public Types</h2></td></tr>
<tr class="memitem:a582c2fb9fe095dbcd1dd24b693b657ac"><td class="memItemLeft" align="right" valign="top">using&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a582c2fb9fe095dbcd1dd24b693b657ac">value_type</a> = std::int8_t</td></tr>
<tr class="separator:a582c2fb9fe095dbcd1dd24b693b657ac"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a6be3619bdb4a919f1510cdc4775bdc52"><td class="memTemplParams" colspan="2">template&lt;typename data_input , typename positions_input &gt; </td></tr>
<tr class="memitem:a6be3619bdb4a919f1510cdc4775bdc52"><td class="memTemplItemLeft" align="right" valign="top">&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a6be3619bdb4a919f1510cdc4775bdc52">VariantMatrix</a> (data_input &amp;&amp;data_, positions_input &amp;&amp;positions_, const std::int8_t max_allele_value=-1)</td></tr>
<tr class="memdesc:a6be3619bdb4a919f1510cdc4775bdc52"><td class="mdescLeft">&#160;</td><td class="mdescRight">"Perfect-forwarding" constructor.  <a href="#a6be3619bdb4a919f1510cdc4775bdc52">More...</a><br /></td></tr>
<tr class="separator:a6be3619bdb4a919f1510cdc4775bdc52"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2763448414490b71461fd26878fd54c2"><td class="memItemLeft" align="right" valign="top"><a id="a2763448414490b71461fd26878fd54c2"></a>
std::int8_t &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a2763448414490b71461fd26878fd54c2">get</a> (const std::size_t site, const std::size_t haplotype)</td></tr>
<tr class="memdesc:a2763448414490b71461fd26878fd54c2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get data from marker <code>site</code> and haplotype <code>haplotype</code>. No range-checking is done. <br /></td></tr>
<tr class="separator:a2763448414490b71461fd26878fd54c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a253d159456282127851c03cecbc37a3c"><td class="memItemLeft" align="right" valign="top"><a id="a253d159456282127851c03cecbc37a3c"></a>
const std::int8_t &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a253d159456282127851c03cecbc37a3c">get</a> (const std::size_t site, const std::size_t haplotype) const</td></tr>
<tr class="memdesc:a253d159456282127851c03cecbc37a3c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get data from marker <code>site</code> and haplotype <code>haplotype</code>. No range-checking is done. <br /></td></tr>
<tr class="separator:a253d159456282127851c03cecbc37a3c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa9785c39872e5a989cd9fafc2609dec7"><td class="memItemLeft" align="right" valign="top"><a id="aa9785c39872e5a989cd9fafc2609dec7"></a>
std::int8_t &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#aa9785c39872e5a989cd9fafc2609dec7">at</a> (const std::size_t site, const std::size_t haplotype)</td></tr>
<tr class="memdesc:aa9785c39872e5a989cd9fafc2609dec7"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get data from marker <code>site</code> and haplotype <code>haplotype</code>. std::out_of_range is thrown if indexes are invalid. <br /></td></tr>
<tr class="separator:aa9785c39872e5a989cd9fafc2609dec7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0d50e7e3900144e3d557acdb6bc82409"><td class="memItemLeft" align="right" valign="top"><a id="a0d50e7e3900144e3d557acdb6bc82409"></a>
const std::int8_t &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a0d50e7e3900144e3d557acdb6bc82409">at</a> (const std::size_t site, const std::size_t haplotype) const</td></tr>
<tr class="memdesc:a0d50e7e3900144e3d557acdb6bc82409"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get data from marker <code>site</code> and haplotype <code>haplotype</code>. std::out_of_range is thrown if indexes are invalid. <br /></td></tr>
<tr class="separator:a0d50e7e3900144e3d557acdb6bc82409"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:a94d7be4fd87fe6b111802bcff30fdbf7"><td class="memItemLeft" align="right" valign="top"><a id="a94d7be4fd87fe6b111802bcff30fdbf7"></a>
std::vector&lt; std::int8_t &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a94d7be4fd87fe6b111802bcff30fdbf7">data</a></td></tr>
<tr class="memdesc:a94d7be4fd87fe6b111802bcff30fdbf7"><td class="mdescLeft">&#160;</td><td class="mdescRight">Data stored in matrix form with rows as sites. <br /></td></tr>
<tr class="separator:a94d7be4fd87fe6b111802bcff30fdbf7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa2f59ddb2b363a764f1f5f625b72c5a9"><td class="memItemLeft" align="right" valign="top"><a id="aa2f59ddb2b363a764f1f5f625b72c5a9"></a>
std::vector&lt; double &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#aa2f59ddb2b363a764f1f5f625b72c5a9">positions</a></td></tr>
<tr class="memdesc:aa2f59ddb2b363a764f1f5f625b72c5a9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Position of sites. <br /></td></tr>
<tr class="separator:aa2f59ddb2b363a764f1f5f625b72c5a9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aac5ad15a638ff8f276e1effd123f5e46"><td class="memItemLeft" align="right" valign="top"><a id="aac5ad15a638ff8f276e1effd123f5e46"></a>
std::size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#aac5ad15a638ff8f276e1effd123f5e46">nsites</a></td></tr>
<tr class="memdesc:aac5ad15a638ff8f276e1effd123f5e46"><td class="mdescLeft">&#160;</td><td class="mdescRight">Number of sites in data set. <br /></td></tr>
<tr class="separator:aac5ad15a638ff8f276e1effd123f5e46"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4b05dbb1c529cf5ffe98c17a256332bc"><td class="memItemLeft" align="right" valign="top"><a id="a4b05dbb1c529cf5ffe98c17a256332bc"></a>
std::size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a4b05dbb1c529cf5ffe98c17a256332bc">nsam</a></td></tr>
<tr class="memdesc:a4b05dbb1c529cf5ffe98c17a256332bc"><td class="mdescLeft">&#160;</td><td class="mdescRight">Sample size of data set. <br /></td></tr>
<tr class="separator:a4b05dbb1c529cf5ffe98c17a256332bc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac5298744e4955e1252632bad6a701562"><td class="memItemLeft" align="right" valign="top"><a id="ac5298744e4955e1252632bad6a701562"></a>
const std::int8_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#ac5298744e4955e1252632bad6a701562">max_allele</a></td></tr>
<tr class="memdesc:ac5298744e4955e1252632bad6a701562"><td class="mdescLeft">&#160;</td><td class="mdescRight">Max allelic value stored in matrix. <br /></td></tr>
<tr class="separator:ac5298744e4955e1252632bad6a701562"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-attribs"></a>
Static Public Attributes</h2></td></tr>
<tr class="memitem:a103282aff369b9029ca10912d6393583"><td class="memItemLeft" align="right" valign="top"><a id="a103282aff369b9029ca10912d6393583"></a>
static const std::int8_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1VariantMatrix.html#a103282aff369b9029ca10912d6393583">mask</a> = std::numeric_limits&lt;std::int8_t&gt;::min()</td></tr>
<tr class="memdesc:a103282aff369b9029ca10912d6393583"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reserved value for masked data. <br /></td></tr>
<tr class="separator:a103282aff369b9029ca10912d6393583"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Matrix representation of variation data. </p>
<p>The data structure is a row-major matrix. Variants are represented by 8-bit integers. Negative values represent missing data, thus allowing up to 128 non-missing states per variable site.</p>
<p>Rows are sites, columns are either haplotypes or are populated as necessary in the case of genotype (unphased) data. Storing variable sites in rows places the performance priority on sites over haplotypes.</p>
<p>The only reserved character state is std::numeric_limits&lt;std::int8_t&gt;::min(), which is used to represent masked data. That reserved value is <a class="el" href="classSequence_1_1VariantMatrix.html#a103282aff369b9029ca10912d6393583" title="Reserved value for masked data. ">VariantMatrix::mask</a>, which is a static constant.</p>
<pre class="fragment">\version 1.9.2  </pre> 
<p class="definition">Definition at line <a class="el" href="VariantMatrix_8hpp_source.html#l00022">22</a> of file <a class="el" href="VariantMatrix_8hpp_source.html">VariantMatrix.hpp</a>.</p>
</div><h2 class="groupheader">Member Typedef Documentation</h2>
<a id="a582c2fb9fe095dbcd1dd24b693b657ac"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a582c2fb9fe095dbcd1dd24b693b657ac">&#9670;&nbsp;</a></span>value_type</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">using <a class="el" href="classSequence_1_1VariantMatrix.html#a582c2fb9fe095dbcd1dd24b693b657ac">Sequence::VariantMatrix::value_type</a> =  std::int8_t</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>The value type of the data. Helpful for generic programming </p>

<p class="definition">Definition at line <a class="el" href="VariantMatrix_8hpp_source.html#l00078">78</a> of file <a class="el" href="VariantMatrix_8hpp_source.html">VariantMatrix.hpp</a>.</p>

</div>
</div>
<h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a id="a6be3619bdb4a919f1510cdc4775bdc52"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a6be3619bdb4a919f1510cdc4775bdc52">&#9670;&nbsp;</a></span>VariantMatrix()</h2>

<div class="memitem">
<div class="memproto">
<div class="memtemplate">
template&lt;typename data_input , typename positions_input &gt; </div>
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Sequence::VariantMatrix::VariantMatrix </td>
          <td>(</td>
          <td class="paramtype">data_input &amp;&amp;&#160;</td>
          <td class="paramname"><em>data_</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">positions_input &amp;&amp;&#160;</td>
          <td class="paramname"><em>positions_</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const std::int8_t&#160;</td>
          <td class="paramname"><em>max_allele_value</em> = <code>-1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>"Perfect-forwarding" constructor. </p>
<p>std::invalid_argument will be thrown if data.size() % positions.size() != 0.0. </p>

<p class="definition">Definition at line <a class="el" href="VariantMatrix_8hpp_source.html#l00082">82</a> of file <a class="el" href="VariantMatrix_8hpp_source.html">VariantMatrix.hpp</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li><a class="el" href="VariantMatrix_8hpp_source.html">VariantMatrix.hpp</a></li>
<li><a class="el" href="VariantMatrix_8cc_source.html">VariantMatrix.cc</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="namespaceSequence.html">Sequence</a></li><li class="navelem"><a class="el" href="classSequence_1_1VariantMatrix.html">VariantMatrix</a></li>
    <li class="footer">Generated on Thu Sep 13 2018 15:07:05 for libsequence by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
