<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>libsequence: libsequence revision history</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">libsequence
   &#160;<span id="projectnumber">1.9.5</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',false,false,'search.php','Search');
});
</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('md__home_molpopgen_src_libsequence_REVISION_HISTORY.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="headertitle">
<div class="title">libsequence revision history </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>What this document implies</h2>
<p>This document lists major changes between tags/releases of libsequence. The following caveat is important:</p>
<p>If you see a section below referring to libsequence version X, and no version X exists as a release, then you must consider the info here inaccurate. In such cases, the list of changes may refer to planned changes for a future release which may or may not be implemented yet in either the master or some development branch.</p>
<h2>Moving towards a "2.0" release: ABI compatibility in flux</h2>
<p>Starting with release 1.8.3, libsequence broke ABI compatibility with previous versions. The "soname" of the library is now set to 20:0:0, where it will remain <b>even as future release continue to break the ABI</b>. Basically, from here until 2.0, you should consider backwards-compatibility to be in flux.</p>
<p>However:</p>
<ul>
<li>As a rule, software depending on libsequence will have to be recompiled, not rewritten. The ABI breakage is (mostly...) coming from changing the implementation, not the interface.</li>
<li>If I remove features, they will be marked as deprecated, meaning that they will be removed in 2.0</li>
</ul>
<h3>ISSUES:</h3>
<ol type="1">
<li><a class="el" href="SeqRegexes_8hpp.html" title="various useful rexex-based functions for dealing with data. Declares Sequence::basic_dna_alphabet,Sequence::full_dna_alphabet, Sequence::pep_alphabet, and the function Sequence::validSeq. ">Sequence/SeqRegexes.hpp</a> &ndash; not working. This will not be fixed until GCC supports &lt;regex&gt;. The function is now currently implemented in a non-regex manner, which is lame, but it works.</li>
</ol>
<h2>libsequence 1.9.5</h2>
<ul>
<li>Support for slices of VariantMatrix and AlleleCountMatrix. <a href="https://github.com/molpopgen/libsequence/pull/49">PR49</a></li>
<li>Faster summary statistic calculation. <a href="https://github.com/molpopgen/libsequence/pull/46">PR46</a> <a href="https://github.com/molpopgen/libsequence/pull/47">PR47</a> <a href="https://github.com/molpopgen/libsequence/pull/48">PR48</a></li>
<li>Added <a class="el" href="namespaceSequence.html#a7686343ff2b87b50a42ce782b594cf0e">Sequence::is_different_matrix</a>. <a href="https://github.com/molpopgen/libsequence/pull/45">PR45</a></li>
<li>Fixed a bug in <a class="el" href="group__popgenanalysis.html#gafb7b705c644cf777c00f98ced5583c12" title="Assign a unique label to each haplotype. ">Sequence::label_haplotypes</a>.<a href="https://github.com/molpopgen/libsequence/pull/44">PR44</a></li>
</ul>
<h2>libsequence 1.9.4</h2>
<p>A major step towards 2.0</p>
<ul>
<li>Summary statistics can now be calculated from a VariantMatrix. <a href="https://github.com/molpopgen/libsequence/pull/33">PR33</a> <a href="https://github.com/molpopgen/libsequence/pull/36">PR36</a></li>
<li>More feature deprecation. See <a href="https://github.com/molpopgen/libsequence/issues/21">Issue 21</a>.</li>
<li>Dependency on TBB removed. <a href="https://github.com/molpopgen/libsequence/pull/38">PR38</a></li>
<li>Dependency on zlib removed. <a href="https://github.com/molpopgen/libsequence/pull/39">PR39</a></li>
</ul>
<h2>libsequence 1.9.3</h2>
<ul>
<li>Refactor unit tests to be much faster to compile</li>
<li>Remove dependency on htslib.</li>
<li>The coalescent simulation machinery is no longer compiled or installed.</li>
<li>Mark a lot of code as deprecated</li>
<li>Travis CI is now Linux-only</li>
<li>Add <a class="el" href="classSequence_1_1VariantMatrix.html" title="Matrix representation of variation data. ">Sequence::VariantMatrix</a> and <a class="el" href="structSequence_1_1StateCounts.html" title="Track character state occurrence at a site in a VariantMatrix. ">Sequence::StateCounts</a></li>
</ul>
<p>This release includes the following GitHub PRs: 11, 12, 13, 14, 15, 25, 26, 30, 31, and 32.</p>
<h2>libsequence 1.9.2</h2>
<ul>
<li>Refactor API for nSL/iHS calculations</li>
</ul>
<h2>libsequence 1.9.1</h2>
<ul>
<li>Sequence::SeqException was removed. Exceptions from namespace std are preferred, and are easier to wrap in other languages.</li>
<li><a class="el" href="classSequence_1_1PolySNP.html#a10608e17d8125b4bcb021b6140465242">Sequence::PolySNP::ThetaL</a> throws exception if outrgroup not present</li>
</ul>
<h2>libsequence 1.9.0</h2>
<ul>
<li>Fixed issues with <a class="el" href="classSequence_1_1Comeron95.html" title="Ka and Ks by Comeron&#39;s (1995) method. ">Sequence::Comeron95</a> that made it impossible to allocate on the stack.</li>
<li>Updated threaded implementation of the l-Haf statistic to use TBB.</li>
<li>Weights on stop codons used in Grantham distance calculations is now configurable, and defaults to the max value of a double. Previous library versions arbitrarily used 999.0.</li>
<li>PolySIM::ThetaL now correctly will not include fixed differences in the calculation.</li>
<li>nSL/iHS, H1, H12, H2H1, and haplotype homozygosity statistics are now calculated in parallel.</li>
<li>Sequence::Disequilibrium parallelized.</li>
<li>Intel's TBB is now a dependency.</li>
</ul>
<h2>libsequence 1.8.9</h2>
<p>Major API and ABI changes! This release is a <b>big</b> step towards a "2.0"-style release.</p>
<ul>
<li>Issue #8 fixed</li>
<li><a class="el" href="classSequence_1_1PolyTableSlice.html" title="A container class for &quot;sliding windows&quot; along a polymorphism table. ">Sequence::PolyTableSlice</a> will throw std::runtime_error if input range is not properly sorted</li>
<li>War on "mutable". The use of this keyword has been removed from the library to the best extent possible.</li>
<li>The API for calculations involving codons has been modernized. This includes <a class="el" href="classSequence_1_1Comeron95.html" title="Ka and Ks by Comeron&#39;s (1995) method. ">Sequence::Comeron95</a>, <a class="el" href="classSequence_1_1RedundancyCom95.html" title="Calculate redundancy of a genetic code using Comeron&#39;s counting scheme. ">Sequence::RedundancyCom95</a>, <a class="el" href="structSequence_1_1WeightingScheme2.html" title="abstract interface to weighting schemes when codons differ at 2 positions ">Sequence::WeightingScheme2</a> (and derived types), <a class="el" href="structSequence_1_1WeightingScheme3.html" title="abstract interface to weighting schemes when codons differ at 3 positions ">Sequence::WeightingScheme3</a> (and derived types), <a class="el" href="classSequence_1_1TwoSubs.html" title="Deal with codons differing at 2 positions. ">Sequence::TwoSubs</a>, <a class="el" href="classSequence_1_1ThreeSubs.html" title="Deal with codons differing at all 3 positions. ">Sequence::ThreeSubs</a>, functions in <a class="el" href="PathwayHelper_8hpp.html" title="declarations of Sequence::Intermediates2 and Sequence::Intermediates3 ">Sequence/PathwayHelper.hpp</a></li>
<li><a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a> (and derived types) have been refactored. The fundamental idea is the same, but the API is modernized. IMO, it is still imperfect, and can be further changed to reflect more idiomatic C++11, but that'll have to wait.</li>
<li>Private data members for classes have been hidden using the <a href="https://en.wikipedia.org/wiki/Opaque_pointer">PIMPL idiom</a>. This goes a long way to future-proofing the ABI compatibility of these types against further implementation changes such as bug fixes.</li>
<li>Sequence/SummStats/classic.hpp provides a sneak previous of how summary statistics will work in the future, once the deprecated <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">Sequence::PolySNP</a> and <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a> can finally be removed</li>
</ul>
<h2>libsequence 1.8.8</h2>
<ul>
<li>l-HAF statistic added (<a class="el" href="lHaf_8hpp_source.html">Sequence/SummStats/lHaf.hpp</a>)</li>
<li>Garud et al.'s H1, H2H1, etc. added (<a class="el" href="Garud_8hpp_source.html">Sequence/SummStats/Garud.hpp</a>)</li>
<li>nSL added (<a class="el" href="nSL_8hpp_source.html">Sequence/SummStats/nSL.hpp</a>)</li>
<li>fixed implementation of <a class="el" href="structSequence_1_1invalidPolyChar.html" title="This functor can be used to determine if a range contains characters that the SNP analysis routines i...">Sequence::invalidPolyChar</a>, which was checking the wrong alphabet</li>
<li>Various documentation fixes</li>
<li><a class="el" href="classSequence_1_1FST.html" title="analysis of population structure using  ">Sequence::FST</a> functions shared,Private, and fixed now throw an exception if deme indexes are out of range. Previously, empty return values were sent, which could be confused with there being no sites in a category.</li>
<li>Various code cleanups, esp. removal of commented-out code blocks</li>
<li>The 8-bit encoding stuff has been removed. This was never used in real-world programs, and suffered from some design issues.</li>
<li><a class="el" href="classSequence_1_1PolyTableSlice.html" title="A container class for &quot;sliding windows&quot; along a polymorphism table. ">Sequence::PolyTableSlice</a> has several updates. First, a bug in "fixed-S" windows was identified through unit testing and fixed. The previous version would drop the last window in some cases. This probably didn't affect many people, but the bug was there for years. (In practice, most 'windows' are fixed distance, not fixed no. variable sites, hence my belief that most previous analyses are ok.) A new constructor supports 'chunking' a PolyTable into equal-sized windows (based on number of variable sites). The class no longer contains a data member of type T, which was never necessary anways.</li>
<li>auto_ptr replaced with unique_ptr in <a class="el" href="Comeron95_8cc_source.html">src/Comeron95.cc</a></li>
<li>binning of nSL/iHS statistics is improved, and better handling of non-finite values implemented.</li>
</ul>
<h2>libsequence 1.8.7</h2>
<ul>
<li>Sliding windows over physical distance now require an end position</li>
<li>A new unit test was added to test the new sliding window code</li>
<li>A test un test/UhapsTest.cc was commented out. It is known to fail, but those parts of libsequence are not used yet anywhere else. I'll deal with it later</li>
</ul>
<h2>libsequence 1.8.6</h2>
<ul>
<li>Patch for Issue #7</li>
</ul>
<h2>libsequence 1.8.5</h2>
<ul>
<li>The API for the template classSequence::PolyTableSlice has changed. During the development of <a href="http://github.com/molpopgen/foRward">another project</a>, I ran into some limitations of the previous API that would require some ugly hacks to work around. The new interface is more modern, and the new implementation will be faster for sliding windows over a physical distance.</li>
<li>The library now installs a program called libsequenceConfig, which may be used to check for what library version is installed:</li>
</ul>
<div class="fragment"><div class="line">libsequenceConfig --version</div></div><!-- fragment --><h2>libsequence 1.8.4</h2>
<ul>
<li>Major improvements to the documentation. The doxygen output now contains a detailed tutorial.</li>
<li>fastq stream + fastq unit tests added</li>
<li>move semantics for the <a class="el" href="classSequence_1_1Seq.html" title="Abstract interface to sequence objects. ">Sequence::Seq</a> hierarchy + unit tests</li>
<li>Programs in examples/ are now compiled via "make check"</li>
<li>The build setup for the unit tests is improved</li>
<li>Old-style enums replaced with C++11 enum classes. (These enums are WAY better.)</li>
<li>The enum changes led to many other changes, primarily in the code base dealing wih calculating differences between codons. The new code is much simpler.</li>
<li>A unit test was added to check the calculations of differences bewtween codons.</li>
<li>Kimura80 and Comeron95 will now return a non-signalling NaN for non-finite values. Previous versions returned "999", which was a bit silly, in all honesty.</li>
<li>PolyTable was redefined to publicly inherit from std::pair&lt; std::vector&lt;double&gt;, std::vector&lt;std::string&gt; &gt;. This change represents a major revision to the code. This change gives full access to the underlying data, and possibly even "too much rope". The change also totally wrecks binary compatiblity with previous versions. All unit tests pass.</li>
<li>Routines for coalescent simulation moved from namespace <a class="el" href="namespaceSequence.html" title="The namespace in which this library resides. ">Sequence</a> to namespace <a class="el" href="namespaceSequence_1_1coalsim.html" title="Routines for coalescent simulation. ">Sequence::coalsim</a>.</li>
<li>Support for 4-bit encoding of sequence data and variation data</li>
<li>Sequence::Ptable, introduced in 1.8.3, was a disaster, as it inherited publicly from std::vector, and was therefore a really poor design decision. The code base has been reverted to a design centered around <a class="el" href="namespaceSequence.html#aa52c570782b66af0cddf21ccaa86c768">Sequence::polySiteVector</a> + conversion functions.</li>
<li>Constructor inheritance for <a class="el" href="classSequence_1_1Seq.html" title="Abstract interface to sequence objects. ">Sequence::Seq</a> enabled.</li>
<li>Sequence/util/vectorizer.hpp added, in order to facilitate safe inheritance from std::vector</li>
<li>Issue #6, which was a bug in bamrecord::seq, is fixed</li>
</ul>
<h2>libsequence 1.8.3 (Dec. 5, 2014)</h2>
<ol type="1">
<li>This release <em>breaks binary compatibility</em> with previous releases. Programs depending on libsequence will likely need to be recompiled.</li>
<li>New constructor functions were added to <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a>, which breaks binary compatibility. These new constructors are needed for easier <a href="http://www.rcpp.org/">Rcpp</a> integration. Derived classes have been updated to use these new constructors whenever possible.</li>
<li>Some I/O functions are no longer inline. These include the operator&lt;&lt; and operator&gt;&gt; for <a class="el" href="classSequence_1_1Seq.html" title="Abstract interface to sequence objects. ">Sequence::Seq</a> and <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a>.</li>
<li>The following header files have been removed: Sequence/ensureFloating.hpp (add too much complexity due to metaprogramming), Sequence/preferFloating.hpp (add too much complexity due to metaprogramming), Sequence/RNG/gsl_rng_wrappers.hpp (boost::bind, std::bind, and lambda expressions render these irrelevant. Removing this header gets rid of the dependency on GSL).</li>
<li>classes Comeron95, PolySNP, and FST no longer inherit from boost::noncopyable. The classes now declarte copy constructors and the assignment operator= in terms of the new C++11 keyword "delete". The classes are still not copyable.</li>
<li>All auto_ptr usage has been replaced with unique_ptr</li>
<li>All function from boost that are now part of C++11 have been replaced with the C++11 function calls. This mostly includes replacing boost::bind with std::bind and BOOST_STATIC_ASSERT (and the correspponding use of boost/type_traits.hpp) with the C++11 &lt;type_traits&gt; equivalent</li>
<li>All examples have been updated to C++11, with no dependencies on external libraries like <a href="http://gnu.org/software/gsl">GSL</a>. Importantly, the GSL is still probably a better way to do random numbers than the C++11 "random" header. (I personally don't like the design of the C++ random number system).</li>
<li>Lots of function prototypes have changed. There were some leftover foo(int x) that are now foo(const int &amp; x). When called many times, the latter may be noticeably faster.</li>
<li>A new class, Sequence::Ptable : public std::vector&lt;Sequence::polymorhpicSite&gt; is added. This is a very powerful class for maninpulating polymorphism data in a general way.</li>
<li>The build system has been rebuilt around automake-1.14</li>
<li>SimDataIO.hpp/.cc have been rewritten. The new implementation is better, cleaner, and breaks binary compatibility.</li>
<li>A new namespace, Sequence::IOhelp, is declared in Sequence/IOhelp.hpp. It contains some functions to reduce the tedium of some IO conversions.</li>
<li>A new, faster implementation of Sequence::SimData::read</li>
<li>The configure script how checks for the existence of <a href="http://www.htslib.org">htslib</a></li>
<li>Sequence/bamreader.hpp and Sequence/bamrecord.hpp added for direct reading from bam files. Will only compile if <a href="http://htslib.org">htslib</a> is detected by the configure script</li>
<li>I've performed extensive code auditing using cppcheck. Many issues were resolved. Most were stylistic, but they do make the code more idiomatic.</li>
<li>A unit test suite is starting to take shape in the test/ subdirectory. These test programs depend on the <a href="http://www.boost.org">boost</a> "test" library. Future tests could add future dependencies.</li>
<li>The unit testing revealed a variety of subtle issues in the code base. All of these were in "dark corners" of the library, <em>e.g.</em> functions that have seen little or no use in production programs.</li>
</ol>
<h2>Historical data (copied from the readme.txt previously found at molpopgen.org)</h2>
<p>current version 1.8.0 (Jan 14, 2014). Two files needed #include &lt;sstream&gt; to used std::ostringstream. Previous library versions were failing to compile on some systems due to the missing header.</p>
<p>version 1.7.9 (Dec 12, 2013) Forward declaration of templates in namespace std were removed from <a class="el" href="typedefs_8hpp.html" title="typedefs used by libsequence ">Sequence/typedefs.hpp</a>. The library will now compile on OS X Mavericks.</p>
<p>version 1.7.8 (Nov 8, 2013): Fixed (a possibly 10 year old) bug in <a class="el" href="SingleSub_8cc_source.html">SingleSub.cc</a>. This bug misclassified AGG to CGG changes as q_0 rather than q_2V. This would lead to errors in synonymous/nonsynonymous substitution counting. Programs like gestimator and polydNdS were affected. However, I have run through a lot of my own data, and recorded no differences in output. You have to have had this specific change (or possibly others involving amino acids with 6 codons in the family). The SingleSub code is now vastly simplified and gives correct output (fingers crossed).</p>
<p>Binary compatilibity with previous versions of libsequence is now broken: All exception specifications are removed in anticipation of that language feature being removed/deprecated.</p>
<p>A new file, SimDataIO.hpp has been added, allowing .gz and binary I/O for SimData objects. See <a class="el" href="test__SimDataIO_8cc_source.html">examples/test_SimDataIO.cc</a> for documentation. This now changes how one compiles programs depending on libsequence: you need zlib installed and must include -lsequence -lz (in that order) when compiling programs.</p>
<p>version 1.7.7: never happened. never publicly released</p>
<p>version 1.7.6 (Feb 13, 2013) Fixed inconsistent exception declarations in templates. libsequence is now compilable with the clang compiler. To use: CXX=clang++ CC=clang ./configure; make ; sudo make install</p>
<p>version 1.7.5 (Dec. 18 2012). The operator&gt;&gt; for Clustalw was not processing whitespace properly when compiled with new versions of gcc. That issue is now fixed.</p>
<p>version 1.7.4 (Oct 31, 2011). Replaced all instances of typename ptrdiff_t with std::ptrdiff_t, as gcc 4.6 is picky enough to care.</p>
<p>1.7.3 (Aug 11, 2011). The more efficient version of <a class="el" href="namespaceSequence_1_1Recombination.html#a4d94f494d332f13638905822cb98bdc5" title="Calculate pairwise LD for a Sequence::PolyTable. ">Sequence::Recombination::Disequilibrium</a> failed to set a flag in the return value, leading to duplicated output. Now fixed.</p>
<p>version 1.7.2 (Aug 11, 2011). The bugfix in 1.7.1 didn't really work. Implementation class data weren't fully-copied. That is now fixed</p>
<p>version 1.7.1 (Aug 3, 2011). Fixed bug in samrecord class. Copying of samrecords led to segfaults in previous versions, due to the private pointer-to-implementation-class not getting its data copied to the new object.</p>
<p>version 1.7.0 (December 13, 2010). A bug is fixed in the calculation of the number of states at a SNP when doing LD calculations. For monomorphic sites, the old version had problems. Also, an initial release of sam/bam record parsing is included (Sequence/sam*.hpp)</p>
<p>version 1.6.9 (June 25,2010) A bug is fixed in calculation of haplotype diversity. When there is considerable missing data in a sequence file, this statistic was severely under-estimated. Previous analyses using this statistic on real data should be re run.</p>
<p>version 1.6.8 (March 29, 2009) Implemented a much more efficient means of calculating basic LD statistics. This is not (yet) available through the PolySNP/PolySIM classes yet, but must be accessed via the header file <a class="el" href="Recombination_8hpp.html" title="namespace Sequence::Recombination ">Sequence/Recombination.hpp</a></p>
<p>version 1.6.7 (March 24, 2009) Implemented a safer iteration through coalescent histories, in light of a recent bug in "ms", on which my routine was based. Also implemented a "max marker distance" for LD calculations. NOTE: This version is binary-incompatible with previous releases, meaning any software depending on libsequence should be recompiled.</p>
<p>version 1.6.6 (Feb 4, 2009) Updated to be compile cleanly with gcc 4.3.x.</p>
<p>version 1.6.5 (Nov. 29, 2007) Updated math function calls to C99 standard. Should compile find on Apple's "Leopard" operating system now.</p>
<p>version 1.6.4 (Jun 19, 2007). Fixed bug where haplotype statistics for SNP data were incorrectly calculated in some cases. When alignment columns containing polymorphic sites also contain a mix of upper- and lower-case letters, the extra case was incorrectly detected as an extra haplotype. This affected the following statistics: # haplotypes, haplotype diversity, Wall's statistics, and Hudson's (1987) estimator of 4Nr.</p>
<p>version 1.6.3 (Jun 22, 2006). Several changes. Templates for coalescent simulation should now be much more flexible with respect to random number generators. Also, the new design allows them to work with boost::bind + GSL routines, simplifying life quite a bit. Code for Hudson's Snn test has been added, as well as code for simulating Coop &amp; Griffiths-style trajectories of beneficial alleles.</p>
<p>version 1.6.2 (March 5, 2006). Three bugs fixed. First, in the FST module, when there are lots of populations, the sum of all the population weights was sometimes detected as not summing to 1, when they actually did. The bug was not accounting for numerical precision, which is now fixed. Second, a rare segfault in the sliding window module (<a class="el" href="PolyTableSlice_8hpp.html">PolyTableSlice.hpp</a>) is now fixed. Third, pairwise LD statistics were being calculated incorrectly in the presence of an outgroup. Now fixed.</p>
<p>version 1.6.1 (Jan 6, 2006). New summary statistics added for SNP data. Some example demographic models for coalescent simulations added (see header <a class="el" href="DemographicModels_8hpp_source.html">Sequence/Coalescent/DemographicModels.hpp</a>). Five bugs fixed: 1.) src/CoalescentInitialize.cc: Fixed bug in init_sample. Regions with 0 "sites" now have the last "site" in the chromosome labelled 0, which is correct. Previously, it was labelled 1.</p>
<p>2.) src/HKA.cc: Fixed minor error in calculation of f_hat which was pointed out by Andy Kern. I was taking the mean(f_hat) across loci as the estimate, rather than following equation 5 of the HKA paper closely enough. The effect was small, and estiamtes of theta and T_hat were unaffected or affected only very little, repsectively.</p>
<p>3.) src/PolySNP.cc: Fixed glitch in calculation of ThetaW. When totMuts==false, only the number of bi-allelic SNPs was used in the numerator of ThetaW. Now, the total number of variable sites is used, as intended.</p>
<p>4.) src/Recombination.cc (affects calculation of LD statistics): In the calculation of D', the 11 gamete was ancestral/ancestral, rather than derived/derived. This is now fixed to match the documentation.</p>
<p>5.) src/CoalescentTreeOperations.cc: in function total_time_on_arg, there was a bug leading to an exception always being thrown. This was due to an inequality being tested as &lt; , when &gt; was what was intended.</p>
<p>version 1.6.0 (Oct 15, 2005). In class PolySNP, summary statistics that require outgroup data now return "nan" when the outgroup is not present. Bug fix: when a PolySNP/PolySIM object was constructed from a SimData object that has 0 segregating sites, the value of 0 was returned for number of haplotypes. It now correctly returns 1.</p>
<p>version 1.5.9 (Sept 21, 2005) more efficient placing of mutations on simulated genealogies. class SimData can now be used as a base class. Note: binary compatibility broken w/previous versions!</p>
<p>version 1.5.8 (July 21, 2005) fixed memory leak introduced in 1.5.7</p>
<p>version 1.5.7 (July 19, 2005) more efficient crossover routine for coalescent simulation. Bug fixed in <a class="el" href="classSequence_1_1SimpleSNP.html" title="SNP table data format. ">Sequence::SimpleSNP</a> that lead to some columns mistakenly getting dropped on input.</p>
<p>version 1.5.6 (Jun 15, 2005). No bugfixes. A function in namespace <a class="el" href="namespaceSequence_1_1Alignment.html" title="Routines fundamental to aligned data. ">Sequence::Alignment</a> is no longer recursive. Library source can now be configured to generate cpu-specific optimizations on G4 and G5 powerpc systems running OS X with gcc.</p>
<p>version 1.5.5 (Jun 9, 2005). Bugfix in <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">Sequence::PolySNP</a> where preprocessing was not correctly applied. The effect of the bug was that some results would be incorrect when performed in certain orders. This affected no programs that I have written and distribute, and analysis of simulated data (using <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a>) was unaffected. In short, it shouldn't have hurt anybody. The non-const operator[] of <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a> now sets non_const_access to true (this fixes a minor bug, which again will rarely affect anything because most operations in the library are on const PolyTables).</p>
<p>version 1.5.4 (May 27, 2005). Major new additions. Old coalescent simulation code is removed. New types and functions allowing simulations with recombination added. changes in support for random number generators implemented (boost-based stu ff removed in favor of the GNU scientific library). This release breaks runtime compatibility.</p>
<p>version 1.5.3 (April 21, 2005) bug in PolySNP::ThetaPi() (i.e. calculation of pi for sequence data) where sites in tables where all but 1 individual had missing data were not handled correctly</p>
<p>version 1.5.2 (April 8, 2005) The bugfix in 1.5.1 was incomplete. Now fixed and tested.</p>
<p>version 1.5.1 (April 5, 2005). Bugfix in sliding window code for non-overlapping windows along the DNA sequence. New example code. Cleanup to code base so that .cc files compile without preprocessor macros.</p>
<p>version 1.5.0 (Jan 11, 2005). Several minor bugfixes: implicit typenames fixed so that library now compiles under gcc 3.4, a bug in the documentation of the formula for PolySNP::ThetaPi() is now corrected, &lt;ieeefp.h&gt; is now included in the HKA source code for systems that have that header (i.e. cygwin), and the estimate of the sample variance in &lt;<a class="el" href="descriptiveStats_8hpp.html" title="classes and functions for calculating descriptive statistics of lists of random variables ...">Sequence/descriptiveStats.hpp</a>&gt; is now unbiased (i.e uses n-1 rather than n in the denominator). In addition, there is a policy change in the calculation of several summary statistics in the classes <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">Sequence::PolySNP</a> and <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a>. When a summary statistic is undefined, the value nan (not a number) is now returned.</p>
<p>version 1.4.9 (Dec 3, 2004). Minor bug in HKA code node fixed. That probably affected nobody, as I don't distribute any programs that depend on it. In addition, several issues with compiling under gcc 3.4 were fixed. These were not related to algorithms, etc., but rather involed non-standard declarations and such.</p>
<p>version 1.4.8 (Nov 3, 2004) &ndash; Major bug in counting derived states fixed in <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">Sequence::PolySNP</a>. Analyses using previous versions of the library should be repeated. Analysis of coalescent sim data (using <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a>) was not affected.</p>
<p>version 1.4.7 (Oct 7, 2004) Class <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a> now correctly sets both Wall's B and Q to -1 if there are &lt; 2 segregating sites. Previously, only Wall's B was set correctly</p>
<p>version 1.4.6 (Oct 6, 2004) Fixed bug in HKA calculations where the poly data in species 1 had 0 seg sites. <a class="el" href="classSequence_1_1SimpleSNP.html" title="SNP table data format. ">Sequence::SimpleSNP</a> now does better checking of its input. PolyTableSlice (aka "sliding windows") now stores windows both with and without segregating sites.</p>
<p>version 1.4.5 (Aug 22, 2004). Bugs fixed: The output shift operator for template class AlignStream now takes a const reference. The input routines for SimParams now reserve memory, which seems to have fixed a rare crash on OS X systems.</p>
<p>New features: <a class="el" href="HKA_8hpp.html" title="Calculations related to the HKA test. ">Sequence/HKA.hpp</a> provides an interface to calculating the HKA test of Hudson, Kreitman, and Aguade.</p>
<p>The class <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a> (<a class="el" href="PolySIM_8hpp.html" title="declaration of Sequence::PolySIM, a class to analyze coalescent simulation data ">Sequence/PolySIM.hpp</a>) is now much more efficient, due to a reorganization of code in the base class <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">Sequence::PolySNP</a> (<a class="el" href="PolySNP_8hpp.html" title="declaration of Sequence::PolySNP, a class to analyze SNP data ">Sequence/PolySNP.hpp</a>). In some applications, calculations can be up to 1/3 faster. The reorganization of code breaks binary compatibility with previous library versions.</p>
<p><a class="el" href="classSequence_1_1SimParams.html" title="Parameters for Hudson&#39;s simulation program. ">Sequence::SimParams</a> (<a class="el" href="SimParams_8hpp.html" title="Sequence::SimParams reads in the parameters of Dick Hudon&#39;s coalescent simulation program...">Sequence/SimParams.hpp</a>) now has a member function fromfile(FILE *), mimicing that of <a class="el" href="classSequence_1_1SimData.html" title="Data from coalescent simulations. ">Sequence::SimData</a>.</p>
<p>version 1.4.4 (Jul 8, 2004) : minor bugfix. All files that use toupper now include &lt;cctype&gt;</p>
<p>version 1.4.3 (Jun 4, 2004) :</p><ul>
<li>bugfixes : in <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>, NumSingletons and NumExternalMutations could give incorrect results with missing data (in cases were all but 1 individual is missing). This affected the output of Fu &amp; Li statistics.</li>
<li>class SimData now does output exactly like hudson's ms program when there are 0 segregating sites</li>
<li>class SimData can now be read in from C-style FILE *.<ul>
<li>made implementations of routines for coalescent simulation a little more robust to uniform random number generators that return 1.0</li>
</ul>
</li>
<li>in <a class="el" href="Alignment_8hpp.html" title="Declaration of namespace Sequence::Alignment. ">Alignment.hpp</a>, an exception specification in a declaration did not match the definition. Fixed.</li>
</ul>
<p>version 1.4.2 (May 14, 2004) &ndash; several bug fixes &amp; new features: *removal of unused variables in several place *efficiency improvement in <a class="el" href="namespaceSequence.html#a0ae39379bc92ab7feed829028035fce9">Sequence::NumDiffs()</a> (in <a class="el" href="Comparisons_8hpp.html" title="delcaration of routines for comparing DNA sequences This file declares a set of functions useful for ...">Comparisons.hpp</a>) *updated Sequence::uniform (in BoostRandomNumbers.hpp) so that it compiles and is compatible with std::random_shuffle *new functions added to namespace Alignment and <a class="el" href="PolyTableFunctions_8hpp.html" title="Operations on non-const Sequence::PolyTable objects. ">PolyTableFunctions.hpp</a> *added several new headers providing template functions for simple statistics *fixed some bugs in <a class="el" href="classSequence_1_1SimpleSNP.html" title="SNP table data format. ">Sequence::SimpleSNP</a> (sample size now printed correctly upon output when there is an outgroup) *fixed bug in Recombination::Disequilibrium (frequency filter now works, case-insensitivity implemented). And, for D and D', the 11 gamete type is in terms of the minor (or derived) allele *fixed use of assert() to deal with compiler warnings</p>
<p>version 1.4.1 (Apr 26, 2004) &ndash; Many bug fixes!</p><ul>
<li>PolySNP &ndash; member functions explicitly check for gaps in snps<ul>
<li>PolySNP &ndash; NumExternalMutations() was incorrect, leading to bad values for Fu/Li statistics with outgroup. Fixed.</li>
</ul>
</li>
<li>PolySNP/PolySIM &ndash; bugs fixed in calculations of Wall's statistics. Q was simply incorrect, and all statistics were incorrect for polymorphism tables constructed with invariant ingroup data</li>
<li>SimpleSNP &ndash; fixed segfault on input for files containing outgroup data<ul>
<li><a class="el" href="PolyTableFunctions_8hpp.html" title="Operations on non-const Sequence::PolyTable objects. ">PolyTableFunctions.hpp</a> added, contains functions to manipulate PolyTable objects in non-const ways</li>
</ul>
</li>
</ul>
<p>version 1.4.0 (Apr 16, 2004) &ndash;</p><ul>
<li>PolySNP::FuLiD() is now correct, no longer returning -inf<ul>
<li>PolySites::read() is now public<ul>
<li><a class="el" href="PolyFunctional_8hpp.html" title="A collection of function objects for SNP analysis. ">PolyFunctional.hpp</a> is now included</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>version 1.3.9 (Apr 12, 2004) &ndash; version 1.3.8 had a bug where the Translate function printed all the codons it processed to stdout. That's fixed in this version. Also fixed a design flaw where output for PolyTables did not work in const contexts.</p>
<p>version 1.3.8 (Apr 08, 2004) : 2 bug fixes: The first was in Sequence::PolyTable::Binary(), and resulted in the binary-format data not being assigned to the object. The second bug was in <a class="el" href="classSequence_1_1SimData.html" title="Data from coalescent simulations. ">Sequence::SimData</a>, and resulted in "ms output" being read in incorrectly when using C++-style input. The member function SimData::fromstdin(), which uses C-style input for speed, however, worked just fine. Other new features include :</p><ul>
<li>non-const operator[] added to PolyTable</li>
<li>public memeber assign() added to AlignStream</li>
<li>print() defined for Clustalw</li>
<li>removed Hudson's code (since in practice everyone just pipes the results of ms to another program anyways...)</li>
<li>libsequence should now be case-preserving of input data</li>
<li>PolyTable now has assign() member functions, to mimic std::vector</li>
<li>PolyTableSlice, a template class for sliding window analyses, is completely re-written and easier to use</li>
<li>updates to documentation</li>
</ul>
<p>version 1.3.7 (Feb 11, 2004) : <a class="el" href="classSequence_1_1SimData.html" title="Data from coalescent simulations. ">Sequence::SimData</a> has 2 changes. First: you can now say while(cin &gt;&gt; d) {} and it works as expected, where d is an instance of SimData. Two, SimData::fromstdin() now returns the return value from the C function fscanf, which makes checking for end of input more robust. Also, the class Sequence::uniform in &lt;Sequence/BoostRandomNumbers.hpp&gt; had it's declaration changed to be compatible with old and new verions of boost.</p>
<p>version 1.3.6 (Feb 02, 2004) : new constructor for PolyTable, allowing construction from a range of PolyTable::const_site_iterator. New classes in &lt;<a class="el" href="shortestPath_8hpp.html" title="Routines to find the shortest distance between any 2 codons, using Grantham&#39;s distance. Declares the class Sequence::shortestPath, and the functions Sequence::mutsShortestPath and Sequence::diffType. ">Sequence/shortestPath.hpp</a>&gt; to aid in dealing w/differences between codons.</p>
<p>version 1.3.5 (Jan 19, 2004) &ndash; Bugfix: PolySNP::Minrec() was returning a value for the Rmin statistic that was incorrect for a small number of data sets. This has been fixed. Other changes : random number system added, based on boost classes. Autoconf macros now work properly at creating #define's for needed headers.</p>
<p>version 1.3.4 (Jan 12, 2004) &ndash; Binary compatibility broken w/previous version. PolyTable column iterators added. New function PolyTable::RemoveAmbiguous added. Ka/Ks code cleaned up. See ChangeLog and manual for details. <a class="el" href="classSequence_1_1Seq.html" title="Abstract interface to sequence objects. ">Sequence::Seq</a> now inherits from std::pair&lt;std::string,std::string&gt;, and various template functions were updated to accomodate that. Template memebers added to <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a>. Requires boost (<a href="http://www.boost.org">http://www.boost.org</a>) to compile.</p>
<p>version 1.3.3 (Nov 1, 2003) &ndash; Exception specifications added. Development headers now included in subdirectory <a class="el" href="namespaceSequence.html" title="The namespace in which this library resides. ">Sequence</a> (so you now need to say things like (#include &lt;<a class="el" href="Fasta_8hpp.html" title="Declaration of Sequence::Fasta streams. ">Sequence/Fasta.hpp</a>&gt;).</p>
<p>version 1.3.2 (August 7, 2003) &ndash; Yesterday's release forgot to include the iterator types for <a class="el" href="classSequence_1_1AlignStream.html" title="Virtual interface to alignment streams. ">Sequence::AlignStream</a>&lt;T&gt;. Here they are. The soname of the shared library is unchanged, since no one downloaded 1.3.1 yet :).</p>
<p>version 1.3.1 (August 6, 2003) &ndash; updated to autoconf 2.5. added new iterator type to <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a>. soname changed to 12:0:0. <a class="el" href="classSequence_1_1FST.html" title="analysis of population structure using  ">Sequence::FST</a> updated.</p>
<p>version 1.3.0 (July 9, 2003) &ndash; Bugfix for <a class="el" href="namespaceSequence.html#a0ae39379bc92ab7feed829028035fce9">Sequence::Comparisons::NumDiffs()</a>. Function now skips missing data as it should. The bug was due to a rewrite of the fxn several versions ago.</p>
<p>version 1.2.9 (Jun 21, 2003) &ndash; Bugfix for PolySNP::ThetaW(). Caused the denominator to be incorrectly calculated. It only occured when running my 'compute' program (from analysis 0.4.8) on several files at once. May have been gcc-3.3 specific, but now it works more generally.</p>
<p>version 1.2.8 (Jun 1, 2003) &ndash; no major changes, so no particular need to update to this version. Restores compatibility with gcc 2.95</p>
<p>version 1.2.7 (Jun 1, 2003) &ndash; added label(unsigned) to <a class="el" href="namespaceSequence.html#a42a51450834a1485455e8b6446415167">Sequence::Hudson2001</a>. Also fixed a preprocessor bug that prevented SeqUtilitites.hpp from being included. No programs were affected by that bug. Also cleaned up example code.</p>
<p>version 1.2.6 (May 21, 2003) &ndash; Two things. One, a bug in PolySNP::ThetaH is fixed which resulted in the value of the statistic being about 6 times too large. (Calculation of ThetaH for simulated was fine). Second, several substantial speed improvements were made to class PolySNP, which speed up the analysis of sequence data. First, calculation of Rmin is no longer recursive, which led to a 10-fold speed up on my machine. Second, a preprocessing of the frequency spectrum now allows summary stastics to be calculated in linear time (at the cost of a bit more memory).</p>
<p>version 1.2.5 &ndash; fixed missing #include&lt;cassert&gt; in <a class="el" href="Comeron95_8cc_source.html">Comeron95.cc</a></p>
<p>version 1.2.4 &ndash; Several bugfixes. PolySNP::WallStats went beyond array bounds for the number of segsites &lt; 2. This is now fixed, and -1 is returned for the statistics, indicating not applicable. Also, several missing #include directives were added to source files, which should make compilers more happy (don't know why gcc 3.3 wasn't complaining, but 2.95 did complain).</p>
<p>version 1.2.3 &ndash; Bugfix release for OS X systems. Sequence::PolyTable::Assign(double *, unsigned, char **, unsigned) ran into an issue where the c++ compiler on OS X does not seem to generate standard-compliant code for vector&lt;T&gt;::assign(). A workaround was implemented.</p>
<p>version 1.2.2 &ndash; Template specializations of routines in namespace <a class="el" href="namespaceSequence_1_1Alignment.html" title="Routines fundamental to aligned data. ">Sequence::Alignment</a> are now provided for std::string. A bug causing a segfault was fixed in Sequence::Alignment::RemoveGaps&lt;T&gt;()</p>
<p>version 1.2.1 &ndash; loads of changes. Sequence::Poly is removed, replaced by <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">Sequence::PolySNP</a> and <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">Sequence::PolySIM</a>. <a class="el" href="classSequence_1_1FST.html" title="analysis of population structure using  ">Sequence::FST</a> is added. see ChangeLog for details</p>
<p>version 1.2.0 &ndash; lots of changes, mostly internal to the code. See ChangeLog.</p>
<p>version 1.1.9 &ndash; lots of changes. Iterators added to Seq and PolyTable. See ChangeLog for details.</p>
<p>version 1.1.8 &ndash; Minimum # of recombination event and disequilibrium coefficients now work with sequence data.</p>
<p>version 1.1.7 &ndash; many changes, see ChangeLog for Details</p>
<p>version 1.1.6 &ndash; fixed bug in input routine for Hudson2001, resulting in occasional segfaults</p>
<p>version 1.1.5 &ndash; fixed bugs in PolyTable::RemoveMultiHits(),ApplyFreqFilter(), and RemoveMissing() that led to segfaults. Also, members of Poly that returned int now return unsigned int</p>
<p>version 1.1.4 &ndash; general code cleanup. Also, removed a simplification in the calculation of Sequence::Poly::ThetaW(SimData *)</p>
<p>version 1.1.3 &ndash; added Sequence::MS_Interface::empty()</p>
<p>version 1.1.2 &ndash; fixed simple bug in output of <a class="el" href="classSequence_1_1SimParams.html" title="Parameters for Hudson&#39;s simulation program. ">Sequence::SimParams</a></p>
<p>version 1.1.1 &ndash; lots of new stuff added to PolyTable. OS X compatibility with g++ 3.1 is now working!</p>
<p>version 1.0.9 &ndash; fixed bug in <a class="el" href="classSequence_1_1PolySites.html#a9a8d2b69228a05251086967e948844ef" title="output a tab-delimited array of positions and character states ">Sequence::PolySites::print()</a> where position of first seg site was off by 1</p>
<p>version 1.0.8 &ndash; D' added back to <a class="el" href="namespaceSequence_1_1Recombination.html#a4d94f494d332f13638905822cb98bdc5" title="Calculate pairwise LD for a Sequence::PolyTable. ">Sequence::Recombination::Disequilibrium()</a></p>
<p>version 1.0.7 &ndash; removed D' statistic from <a class="el" href="namespaceSequence_1_1Recombination.html#a4d94f494d332f13638905822cb98bdc5" title="Calculate pairwise LD for a Sequence::PolyTable. ">Sequence::Recombination::Disequilibrium</a> until a correct version of the code can be written (I found an error in the calculation and I don't have time to fix it right now)</p>
<p>version 1.0.6 &ndash; generalized Poly::ThetaW(PolyData) to handle missing data. fixed several documentation bugs.</p>
<p>version 1.0.5 &ndash; fixed a problem in <a class="el" href="SimData_8cc_source.html">SimData.cc</a> that prevented compiling on Sun systems (did not affect output, just compiling)</p>
<p>version: 1.0.4 &ndash; fixed bug in processing simulation output when segstites = 0. Also added a new method to <a class="el" href="classSequence_1_1SimData.html" title="Data from coalescent simulations. ">Sequence::SimData</a> that really speeds up input, see manual for details</p>
<p>version: 1.0.3 &ndash; stable, things work well :) NOTE: a bug has been found in <a class="el" href="classSequence_1_1SimData.html" title="Data from coalescent simulations. ">Sequence::SimData</a> that causes a segfault when there are 0 segregating sites in the output. Fix coming soon. </p>
</div></div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Thu Sep 13 2018 15:07:05 for libsequence by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
